app/models/manageiq/providers/vmware/cloud_manager/vm/operations.rb

```diff
@@ -1,3 +1,23 @@
 module ManageIQ::Providers::Vmware::CloudManager::Vm::Operations
+  extend ActiveSupport::Concern
+
   include_concern 'Power'
+
+  def raw_destroy
+    raise "VM has no #{ui_lookup(:table => "ext_management_systems")}, unable to destroy VM" unless ext_management_system
+    ext_management_system.with_provider_connection do |service|
+      begin
+        response = service.post_undeploy_vapp(ems_ref, :UndeployPowerAction => 'shutdown')
+        service.process_task(response.body)
+      rescue => err
+        $vcloud_log.debug("vm=[#{name}], error: VM not running! Ignoring error: #{err}")
+      ensure
+        response = service.delete_vapp(ems_ref)
+        service.process_task(response.body)
+        update_attributes!(:raw_power_state => "terminated")
```

Comment:
I think correct power state for deleted VM should be "terminated". Maybe I should set power state to  "unknown" instead of "terminated"?